include/seqan3/io/sequence_file/format_fasta.hpp

```diff
@@ -191,20 +182,71 @@ class sequence_file_format_fasta
         {
             if (options.truncate_ids)
             {
-                std::ranges::copy(stream_view | std::view::drop_while(is_id || is_blank)     // skip leading >
+            #if SEQAN3_WORKAROUND_VIEW_PERFORMANCE
+                auto it = stream_view.begin();
+                auto e = stream_view.end();
+                for (; (it != e) && (is_id || is_blank)(*it); ++it)
+                {}
+
+                bool at_delimiter = false;
+                for (; it != e; ++it)
+                {
+                    if ((is_cntrl || is_blank)(*it))
+                    {
+                        at_delimiter = true;
+                        break;
+                    }
+                    id.push_back(assign_char_to(*it, value_type_t<id_type>{}));
+                }
+
+                if (!at_delimiter)
+                    throw unexpected_end_of_input{"FastA ID line did not end in newline."};
+
+                for (; (it != e) && ((!is_char<'\n'>)(*it)); ++it)
+                {}
+
+            #else // ^^^ WORKAROUND | ORIGINAL ↓↓↓
```

Comment:
Shouldn't we not stick to ASCII characters whenever possible?

Reply:
Well, we defined the following rule in the wiki ages ago:
```
 All files are expected to be UTF-8 encoded. Special characters are allowed in documentation and string literals, but not in regular code (function names etc) and file names.
```

I use UTF8 in a lot of places...